packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke.ts

```diff
@@ -0,0 +1,277 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for invoking an API Endpoint with ApiGatewayInvoke
+ */
+export interface ApiGatewayInvokeProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * hostname of an API Gateway URL
+   * @example {ApiId}.execute-api.{region}.amazonaws.com
+   */
+  readonly apiEndpoint: string;
+
+  /**
+   * Http method for the API
+   */
+  readonly method: HttpMethod;
+
+  /**
+   * HTTP headers string to list of strings
+   * @default - No headers
+   */
+  readonly headers?: sfn.TaskInput;
+
+  /**
+   * Name of the stage where the API is deployed to in API Gateway
+   * @default - Required for REST and $default for HTTP
+   */
+  readonly stage?: string;
+
+  /**
+   * Path parameters appended after API endpoint
+   * @default - No path
+   */
+  readonly path?: string;
+
+  /**
+   * Query strings string to list of strings
+   * @default - No query parameters
+   */
+  readonly queryParameters?: sfn.TaskInput;
+
+  /**
+   * HTTP Request body
+   * @default - No requestBody
+   */
+  readonly requestBody?: sfn.TaskInput;
+  /**
+   * Authentication methods
+   *
+   * NO_AUTH: call the API direclty with no authorization method
+   *
+   * IAM_ROLE: Use the IAM role associated with the current state machine for authorization
+   *
+   * RESOURCE_POLICY: Use the resource policy of the API for authorization
```

Comment:
this looks it would be more suitable to be an enum. Any reason why we would need it to be of type `TaskInput` aside from the type in the Sfn documentation stating it supports `Json`?